### PR TITLE
Setup container test for ruby

### PR DESCRIPTION
**Pull request by neomatrix369**
- might not have the optimum list of packages in the apt-get list but its best to keep it that way or else we are spending more time than we should
- added a script to help us debug containers by overriding the `entrypoint` command via `/bin/bash`
usage:
```
./debugDockerContainer.sh ruby participant round https://github.com/julianghionoiu/tdl-runner-ruby sync-java-code-coverage-changes-to-ruby SUM
```
When inside the container run:
```
./fetch_repo_and_collect_coverage.sh SUM
```

We haven't applied any flag to ignore failed tests, so that the test-runner does not exit at the first failing test and coverage plugin gets the full picture.

### CODE REVIEW

File: container/images/ruby/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    software-properties-common \
+    python-software-properties \
+    sqlite3 \
+    yarn  \
+    zlib1g-dev \
+    --no-install-recommends && rm -r /var/lib/apt/lists/*
+
+RUN /bin/bash -l -c "echo progress-bar >> ~/.curlrc"
+RUN curl -sSL https://rvm.io/mpapis.asc | gpg --import -
+RUN echo "Downloading and installing 'rvm'" && curl -sSL https://get.rvm.io | bash -s stable
```
Comment (julianghionoiu) [inline container/images/ruby/Dockerfile]: Why do we need both RVM and Rbenv?

File: container/images/ruby/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+RUN echo "Downloading and installing 'rbenv'" && git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+RUN echo "export PATH=$HOME/.rbenv/bin:$PATH" >> ~/.bashrc
+RUN echo 'eval "$(rbenv init -)"' >> $HOME/.bashrc
+RUN git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build
+RUN echo 'export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$PATH"' >> ~/.bashrc
+RUN echo 'export PATH="$HOME/.rbenv/versions/${RUBY_VERSION}/bin:$PATH"' >> ~/.bashrc
+
+RUN /bin/bash -l -c "echo Downloading and installing \'ruby via rbenv\'"
```
Comment (julianghionoiu) [inline container/images/ruby/Dockerfile]: Is there an easier way to install Ruby? Something like using Apt-get? 
It feels a bit too complicated.

File: container/images/ruby/Dockerfile (hunk 1 above)
Comment (neomatrix369) [reply container/images/ruby/Dockerfile]: We need version 2.2.2 as per the project config and I didnt get any success using any other means, rvm and rbenv worked.

Maybe we don't need the patch installation on top of 2.2.2 - on that note is it possible to move to another version which can be installed more easily?

File: container/images/ruby/Dockerfile (hunk 2 above)
Comment (neomatrix369) [reply container/images/ruby/Dockerfile]: I agree its complicated and I had trouble getting 2.2.2 to work as I mentioned above, I will try another route but this was a bit fiddley

Comment (neomatrix369) [reply container/images/ruby/Dockerfile]: Have a look at the new version of the changes

**Pull request merged**